### ISSUE TITLE
[XMaterial] - Matching wrong XMaterial from ItemStack on pre-1.13

### ISSUE THREAD

**Issue by LOOHP** [CryptoMorin/XSeries#75]
**Description**
When using the `XMaterial.matchXMaterial(itemstack);` on versions under 1.12.2, some blocks/items get matched with the wrong XMaterial, during my testing:

The "Grass Block" in pre1.13 is called "Grass", therefore XMaterial matched it with the "Grass" XMaterial, however the correct match should be "Grass_Block", as post1.13 "Grass" is equivalent to pre1.13 "Tall Grass".

Another one is Terracotta blocks, which is called "\<Color\> Hardened Clay" in 1.8 and they got matched with "\<Color\> Glazed Terracotta".

**Version**
I've tested on 1.16.3, 1.12.2 and 1.8.8

**Comment by CryptoMorin** [comment]
I'm really having a hard time understanding these. Yes, it might sound really simple, but if you provided your full code and explained it as a comment line by line, it'd help a lot. I had to read it several times.

First of all, I think you meant to say, `on 1.12 and versions below` or `below 1.13` and not excluding 1.12, since you tested this in 1.12 and the fact that you used terms like `pre1.13` and `post1.13`

For example, you said,
> When using the XMaterial.matchXMaterial(itemstack);

and then
> The "Grass Block" in pre1.13 is called "Grass", therefore XMaterial matched it with the "Grass" XMaterial

First of all, I'm not sure what you're trying to match here exactly. **GRASS_BLOCK** or **GRASS**
According to your previous comment, you used the match method with the `ItemStack` parameter. So I think you meant to match **GRASS** since **GRASS_BLOCK** isn't a material in pre-1.13
> "Grass" is equivalent to pre1.13 "Tall Grass".

Pre-1.13 doesn't even have a material called **TALL_GRASS**, did you mean to say **LONG_GRASS**?
So yes, the code below does indeed return **XMaterial.GRASS** and **Material.GRASS** when parsed instead of **XMaterial.GRASS** and **Material.LONG_GRASS**:
```java
XMaterial.matchXMaterial(new ItemStack(Material.GRASS));
```
I'll fix that.

---
> Another one is Terracotta blocks, which is called "<Color> Hardened Clay" in 1.8 and they got matched with "<Color> Glazed Terracotta".

Okay so here I'm sure you're matching **terracotta** items and not **glazed terracotta** and the fact that glazed terracotta doesn't exist in 1.8
I think you're talking about in-game names here now. There are no materials with the structure `<color>_HARDENED_CLAY` in 1.8 or 1.12 **API**
So to simplify it, matching `<color>_TERRACOTTA` items in 1.8 results in `<color>_GLAZED_TERRACOTTA`
However this bug still happens in 1.12, I'm wondering why you had to test this in 1.8?
<br>
Please correct anything if I'm wrong. Right now I'm really confused.

**Comment by LOOHP** [comment]
Uhh, sorry about that.
I'm not really sure how to make it sound less confusing.

Ok let me try again.

So if I am holding a Grass Block in 1.12 in my first slot.
`ItemStack item = player.getInventory().getItem(0);`
Then I try to match it to an XMaterial
`XMaterial.matchXMaterial(item);`
It will return
`XMaterial.GRASS` instead of `XMaterial.GRASS_BLOCK`

Also if I hold any color of Hardened Clay in 1.12 in my first slot.
Then try to match it with the same code above.
It returns the corresponding color of a `XMaterial.<Color>_GLAZED_TERRACOTTA` instead of normal terracotta.

Is that a bit less confusing?
